jupyterhub/handlers/base.py

```diff
@@ -936,6 +936,17 @@ def _clear_spawn_future(f):
 
         finish_spawn_future.add_done_callback(_clear_spawn_future)
 
+
+        # when spawn fails with an exception
+        # update prometheus metrics
+        def _metrics_server_spawn_failed(f):
+            if f.exception():
+                SERVER_SPAWN_DURATION_SECONDS.labels(
+                    status=ServerSpawnStatus.failure
+                ).observe(time.perf_counter() - spawn_start_time)
+
+        finish_spawn_future.add_done_callback(_metrics_server_spawn_failed)
```

Comment:
Instead of registering a new callback, lets update the metric in ` _track_failure_count`  (around [this line](https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/handlers/base.py#L947)).
This way, we also handle the case when the spawn gets cancelled.

What do you think?

Reply:
Yep, this is much cleaner! Updated the change.